### DISCUSSION TITLE
Allow execution in batch mode

### DISCUSSION THREAD

**Discussion by codie3611** [copier-org/copier#447] [Ideas]
Hi, I would like to use copier from another program but struggle with the fact that it is designed to be interactive. For example the answers cannot be handed over directly in shell-mode. How easy would it be to realize something like this?

**Comment by pawamoy** [comment]
Do you want to call Copier from a Python process? See https://github.com/copier-org/copier/blob/master/copier/main.py#L747. You can definitely pass data to avoid the interactive prompts. If you want to call it as a subprocess (be it in a shell-script, a Python process, or any other executable), you can as well pass data with the `-d` switch, example here: https://copier.readthedocs.io/en/latest/configuring/#data, all flags here: https://copier.readthedocs.io/en/latest/reference/cli/#copier.cli (warning: latest docs, install the latest prerelease version).

**Reply by codie3611** [reply]
Thanks for the reply, I used the `-d` switch but it still asked me the question as I did not realize this is what the `-f` flag is for. I will try your recommendation but would already suggest to make this more clear.

**Comment by yajo** [comment]
Please feel free to open a PR and improve the docs. Thanks!
